components/text_recognition/browser/text_recognition_helper_win.cc

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/text_recognition/browser/text_recognition_helper_win.h"
+
+#include <utility>
+
+#include "base/logging.h"
+#include "base/task/task_traits.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/text_recognition/browser/text_recognition.h"
+#include "content/public/browser/browser_thread.h"
+
+namespace text_recognition {
+
+TextRecognitionHelperWin::TextRecognitionHelperWin() = default;
+TextRecognitionHelperWin::~TextRecognitionHelperWin() = default;
+
+void TextRecognitionHelperWin::GetTextFromImage(
+    const SkBitmap& image,
+    base::OnceCallback<void(const std::vector<std::string>&)> callback) {
+  callback_ = std::move(callback);
+  image_ = image;
+
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE,
+      {base::MayBlock(), base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN},
+      base::BindOnce(&GetAvailableRecognizerLanguages),
+      base::BindOnce(&TextRecognitionHelperWin::OnGetAvailableLanguages,
+                     weak_factory_.GetWeakPtr()));
+}
+
+void TextRecognitionHelperWin::SetResultText(
+    const std::vector<std::string>& text) {
+  if (text.empty()) {
+    return;
+  }
+
+  if (best_result_.empty()) {
+    best_result_ = text;
+  }
+
+  // Determine which one is more better result between |text| and
+  // |best_result_|. We choose one that has more longer detected string.
```

Comment:
just curious if this is valid approach, languages are different, is this recommended by some docs?

Reply:
I also don't know it's the best approach to pick best result from multiple detected text.
https://learn.microsoft.com/en-us/uwp/api/windows.media.ocr.ocrengine?view=winrt-22621 is the only doc what I found.
`OcrEngine.RecognizeAsync` only detect text for specified language.
As we can't know the language in the image in advance, I think we just need to try to detect with all supported languages. And then we could pick one result from them.
I thought most longest result would be best pick.